pages/blog/_posts/2024-08-16-incident-report.mdx

```diff
@@ -0,0 +1,74 @@
+
+We apologize for incurring this downtime for all Inngest customers. System reliability is of the utmost importance for us, and we will ensure this type of failure does not happen again.
+
+Full timeline, root cause and corrective actions are included below with additional detail.
+
+## Timeline
+
+- **2024-08-16 00:05 UTC** - SDK Gateways stop fulfilling requests. Function execution is failing.
+- **2024-08-16 00:10 UTC** - On-call engineers are paged with an alert citing high volume of function execution failures.
+- **2024-08-16 00:25 UTC** - Issue is pinpointed to the SDK Gateways. Team attempts to restart instances to reset state.
+- **2024-08-16 00:38 UTC** - Executor services are scaled down to zero to prevent thundering herd issues when SDK Gateways are brought back online.
+- **2024-08-16 00:51 UTC** - Manual intervention to clear disk space begins. Internal service discovery IPs are rotated. **Function execution is partially restored.**
+- **2024-08-16 01:01 UTC** - All SDK Gateways are back online. New public IPs are issued.
+- **2024-08-16 01:51 UTC** - One SDK Gateway is found not to be functioning correctly and is rebooted and service discovery IPs are rotated.
+- **2024-08-16 04:40 UTC** - Networking issues reported by Vercel customers.
+- **2024-08-16 04:47 UTC** - New static IP addresses created and process of configuration begins. New static IPs are published to our IP v4 public list for partners.
+- **2024-08-16 05:30 UTC** - New SDK Gateways with static IPs start to roll out.
+- **2024-08-16 05:45 UTC** - All SDK Gateways are rolled out. **Function execution is fully restored**.
+- **2024-08-16 06:02 UTC** - Additional SDK Gateways are added for extra capacity and redundancy.
+
+## Root Cause
+
+The SDK Gateway forward proxy handles all egress HTTPS traffic from our function “executor” service to ensure consistent IP addresses are used for outbound traffic to our various cloud provider partners, such as Vercel. Our “executors” are responsible for pulling work off internal queues to invoke user functions at whatever URL they may be hosted at. These executors run within our Kubernetes cluster, scale depending on load, and IPs are ephemeral, so the SDK Gateway is used as a proxy for outbound requests. The SDK Gateway runs a group of machines, each with static IPs assigned.
+
+![System architecture showing executors and SDK Gateway](/assets/blog/2024-08-15-incident-report/architecture.png)
+
+For audit purposes, the SDK Gateway logs details on every request it makes.  Due to increased throughput beginning the morning of Thursday 15th, instances started running our of disk space.  The lack of disk space prevented logging, which caused the processes running on the instances to hang. Log rotation was not properly configured on these instances. Fixing this issue immediately require manual intervention. Log rotation has since been configured to run hourly on these instances.
```

Comment:
typo

```suggestion
For audit purposes, the SDK Gateway logs details on every request it makes.  Due to increased throughput beginning the morning of Thursday 15th, instances started running out of disk space.  The lack of disk space prevented logging, which caused the processes running on the instances to hang. Log rotation was not properly configured on these instances. Fixing this issue immediately require manual intervention. Log rotation has since been configured to run hourly on these instances.
```